quickwit/quickwit-indexing/src/source/file_source.rs

```diff
@@ -140,114 +122,59 @@ impl TypedSourceFactory for FileSourceFactory {
         source_runtime: SourceRuntime,
         params: FileSourceParams,
     ) -> anyhow::Result<FileSource> {
-        let checkpoint = source_runtime.fetch_checkpoint().await?;
-        let mut offset = 0;
-
-        let reader: FileSourceReader = if let Some(filepath) = &params.filepath {
+        let file_source = if let Some(filepath) = &params.filepath {
             let partition_id = PartitionId::from(filepath.to_string_lossy().borrow());
-            offset = checkpoint
+            let checkpoint = source_runtime.fetch_checkpoint().await?;
+            let offset = checkpoint
                 .position_for_partition(&partition_id)
                 .map(|position| {
                     position
                         .as_usize()
                         .expect("file offset should be stored as usize")
```

Comment:
I know you are not resposnible for the previous code, but could we return an error here?